### ISSUE TITLE
Adding DIoU loss and CIoU Loss to keras-cv

### ISSUE THREAD

**Issue by Anirban0011** [keras-team/keras-cv#1736]
Should this two losses be added to keras_cv.losses ?

1. DIoU (Distance over IoU) Loss is independent to the size of the object being detected. It is seen to converge mush faster than GIoU even with opposite orientation of boxes(vertical and horizontal). It is also deployed as a criterion in NMS layer of which we already have an implementation in the repo.

   * paper link : https://arxiv.org/abs/1911.08287
   * implementation link : https://github.com/Zzh-tju/DIoU

2. CIoU (Complete IoU) loss needs fewer iteration than both GIoU and DIoU loss to converge. It has been deployed in YOLO v3, 
    Yolo v4, SSD, and Faster RCNN.
   
     * paper link (same) : https://arxiv.org/abs/1911.08287
     * implementation link : https://github.com/Zzh-tju/CIoU

**Comment by Anirban0011** [comment]
A paper which highlights the limitations of existing BBR loss functions : [Focal and Efficient IOU Loss for Accurate Bounding Box Regression](https://arxiv.org/abs/2101.08158)

**Comment by jaygala223** [comment]
Hello @ianstenbit, can you please take a look at this issue? I would like to work on this

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new KerasHub package, which supports image, text, and multi-modal models. Please read https://github.com/keras-team/keras-hub/issues/1831. KerasHub will support all the core functionality of KerasCV.

KerasHub can be installed with !pip install -U keras-hub. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).
